File: 프로그래머스/unrated/181935. 홀짝에 따라 다른 값 반환하기/홀짝에 따라 다른 값 반환하기.py
def solution(n):
    if n % 2 == 0:
        answer = [i**2 for i in range(1, n+1) if i % 2 == 0]
    else:
        answer = [i for i in range(1, n+1) if i % 2 != 0]
    return sum(answer)

File: 프로그래머스/0/181935. 홀짝에 따라 다른 값 반환하기/홀짝에 따라 다른 값 반환하기.java
class Solution {
    public int solution(int n) {
        int answer = 0;
        if (n % 2 == 0) {
            // 짝수 일때
            for (int i = 1; i < n + 1; i++) {
                if ( i % 2 == 0) {
                    answer += i * i;
                }
            }
        }
        else {
            // 홀 수 일때
            for (int i = 1; i < n + 1; i++) {
                if ( i % 2 != 0) {
                    answer += i;
                }
            }
        }
        return answer;
    }
}